### PR TITLE
Fixes Issue #401 - Migrating styling to MUI's sx prop

### LINKED ISSUE #401: MUI: use `sx` prop instead of inline `style`
After the upgrade to MUI v5, `sx` should be usable on most components:
https://mui.com/system/the-sx-prop/

### PR DESCRIPTION
**Pull request by ramprakashram**
The current implementation has used _inline styling_ and _makeStyles_(currently deprecated) for styling the components. This PR upgrades them to the newer _sx_ props wherever possible.

Please review and let me know if I need to change anything.

Fixes #401

### CODE REVIEW

File: frontend/src/features/comparisons/ComparisonList.tsx [hunk 1]
```diff
@@ -67,10 +58,18 @@ const Comparisons = ({
...
 
   return (
-    <Container className={classes.content}>
+    <Container
+      sx={{
+        padding: theme.spacing(3),
+        [theme.breakpoints.down('sm')]: {
+          padding: theme.spacing(0),
+        },
+        maxWidth: 840,
```
Comment (GresilleSiffle) [inline frontend/src/features/comparisons/ComparisonList.tsx:70]: The `!important` is needed here to match the previous layout. Without it, the container takes up to 1200px in full screen.

```suggestion
        maxWidth: '840px !important',
```

**Review by GresilleSiffle — CHANGES_REQUESTED** [review]
Hello @ramprakashram 

Thanks again for your work. I made a small recommendation about factorizing a `sx` value, and a recommendation in the comparison list that "fix" a missing `!important`.

File: frontend/src/features/frame/components/topbar/AccountInfo.tsx [hunk 2]
```diff
@@ -54,9 +29,14 @@ const LoggedInActions = () => {
       <Button
         variant="outlined"
         color="inherit"
-        sx={{ borderColor: 'rgba(0, 0, 0, 0.23)' }}
+        sx={{
+          borderColor: 'rgba(0, 0, 0, 0.23)',
```
[the remark this answers is not in the document]
Comment (ramprakashram) [reply frontend/src/features/frame/components/topbar/AccountInfo.tsx]: @GresilleSiffle Updated the _AccountInfo_ component as you suggested.

File: frontend/src/features/comparisons/ComparisonList.tsx (hunk 1 above)
Comment (ramprakashram) [reply frontend/src/features/comparisons/ComparisonList.tsx]: @GresilleSiffle Updated the _ComparisonList_ component as you suggested.

File: frontend/src/features/videos/VideoCard.tsx [hunk 3]
```diff
@@ -264,12 +238,18 @@ function VideoCard({
...
+      sx={{
+        ...mainSx(theme),
+      }}
+    >
       <Grid
         item
         xs={12}
         sm={compact ? 12 : 4}
-        style={{ aspectRatio: '16 / 9', padding: 0 }}
+        sx={{ aspectRatio: '16 / 9', padding: 0 }}
```
Comment (amatissart) [inline frontend/src/features/videos/VideoCard.tsx:252]: Here too, using `sx` creates a conflict between our custom style, and the defaults defined by MUI.
I am not too fond of using `!important`, but at least it makes explicit that there is a conflict here, and that we may want to update this `<Grid>` layout in the future.

```suggestion
        sx={{ aspectRatio: '16 / 9', padding: '0 !important' }}

```

File: frontend/src/features/videos/VideoCard.tsx [hunk 4]
```diff
@@ -264,12 +238,18 @@ function VideoCard({
   }
 
   return (
-    <Grid container spacing={1} className={classes.main}>
+    <Grid
+      container
+      spacing={1}
+      sx={{
+        ...mainSx(theme),
+      }}
```
Comment (amatissart) [inline frontend/src/features/videos/VideoCard.tsx:246]: `sx` accepts a callback, and other "theme-aware" properties: https://mui.com/system/the-sx-prop/#theme-aware-properties
```suggestion
    <Grid container spacing={1} sx={mainSx}>
```

**Review by amatissart — CHANGES_REQUESTED** [review]
One more remaining issue in the `VideoCard`

File: frontend/src/features/videos/VideoCard.tsx (hunk 3 above)
Comment (ramprakashram) [reply frontend/src/features/videos/VideoCard.tsx]: @amatissart Updated the _VideoCard_ component as you suggested.

File: frontend/src/features/videos/VideoCard.tsx (hunk 4 above)
Comment (ramprakashram) [reply frontend/src/features/videos/VideoCard.tsx]: @amatissart Thanks for letting me know, I just updated the two components (AccountInfo and VideoCard) in which we are using Theme Object directly.

**Comment by GresilleSiffle** [comment]
LGTM now.

Any suggestion @amatissart or should we merge?

**Review by GresilleSiffle — APPROVED** [review]

**Review by amatissart — APPROVED** [review]
I just applied a few more corrections when reviewing the branch with @GresilleSiffle.
It should be ok now.
Thanks @ramprakashram

**Comment by GresilleSiffle** [comment]
Oops, I just merge another pull request which introduces new conflicts.

**Pull request merged**